tasks/main.yml

```diff
@@ -5,23 +5,149 @@
 - name: Set platform/version specific variables
   include_tasks: tasks/set_vars.yml
 
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: Check if requested version is supported in the system (RHEL8)
+  fail:
+    msg: "RHEL 8 supports only Postgresql 10, 12 and 13"
+  when:
+    - ansible_facts['os_family'] == "RedHat"
+    - ansible_facts['distribution_major_version'] == "8"
+    - postgresql_version not in rhel8_streams
+
+- name: Check if requested version is supported in system (RHEL9)
+  fail:
+    msg: "RHEL 9 supports only Postgresql 13"
+  when:
+    - ansible_facts['os_family'] == "RedHat"
+    - ansible_facts['distribution_major_version'] == "9"
+    - postgresql_version not in rhel9_streams
+
+- name: Check requested and installed version of Postgresql
+  fail:
+    msg: "Setting version {{ postgresql_version }} while
+      {{ ansible_facts.packages['postgresql'][0].version }} is installed"
+  when:
+    - "'postgresql' in ansible_facts.packages"
+    - ansible_facts.packages['postgresql'][0].version |
+      regex_search('([0-9]*).', '\\1') | first !=  postgresql_version
+
+- name: Ensure required packages are installed on RHEL8 based systems
   package:
-    name: "{{ __template_packages }}"
+    name: "@postgresql:{{ postgresql_version }}/server"
     state: present
+  when:
+    - ansible_facts['os_family'] == "RedHat"
+    - ansible_facts['distribution_major_version'] == "8"
+
+# RHEL 9 doesn't support modules yet
+- name: Ensure required packages are installed on RHEL9 based systems and others
+  package:
+    name: "postgresql-server"
+    state: present
+  when:
+    - ansible_facts['distribution_major_version'] != "8"
+
+- name: Init DB and set up password
+  block:
+    - name: Init DB
+      command:
+        cmd: postgresql-setup --initdb
+        creates: /var/lib/pgsql/data/postgresql.conf
+
+    - name: Start Postgresql server
+      service:
+        name: "postgresql"
+        state: started
+        enabled: yes
+
+    - name: Set password for super user
+      become: yes
+      become_user: postgres
+      environment:
+        PGPASSWORD: "{{ postgresql_password }}"
+      command:
+        cmd: >
+          psql -c "ALTER USER postgres WITH ENCRYPTED PASSWORD
+          '{{ postgresql_password }}';"
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
+    - name: Enable logging in by password
+      replace:
+        path: /var/lib/pgsql/data/pg_hba.conf
+        regexp: '(peer|ident)$'
+        replace: 'md5'
+        backup: yes
+      notify: restart postgresql
+  when:
+    - not postgresql_conf.stat.exists
```

Comment:
you can put this on the previous line with `when: not postgresql_conf.stat.exists`